### PR TITLE
add: hw6

### PR DESCRIPTION
**Pull request by stsepchanka**

### CODE REVIEW

File: src/middleware/authorize.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  next: NextFunction
+): void {
+  let token = req.headers.authorization;
+
+  if (!token) {
+    return next(new UnauthorizedError("No token provided"));
+  }
+
+  if (token.startsWith("Bearer ")) {
+    token = token.slice(7, token.length);
```
Comment (kuzanatoliy) [inline src/middleware/authorize.ts]: Magic number

File: src/middleware/authorize.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  res: Response,
+  next: NextFunction
+): void {
+  let token = req.headers.authorization;
+
+  if (!token) {
+    return next(new UnauthorizedError("No token provided"));
+  }
+
+  if (token.startsWith("Bearer ")) {
```
Comment (kuzanatoliy) [inline src/middleware/authorize.ts]: I would use
const [tokenType, tokenValue] = token.split(' ');
if(tokenType === TOKEN_TYPE && tokenValue)
...

File: src/app.ts [hunk 3]
```diff
@@ -1,15 +1,19 @@
...
-import { catchErrors, logInvokedMethods } from "./middleware";
+import cors from "cors";
 
-import { usersRouter, groupsRouter } from "./routers";
+import { authorize, catchErrors, logInvokedMethods } from "./middleware";
+import { loginRouter, usersRouter, groupsRouter } from "./routers";
 
 export const app = express();
 
+app.use(cors());
```
Comment (kuzanatoliy) [inline src/app.ts]: It would be nice if you used more specific cors rules

File: src/middleware/authorize.ts (hunk 2 above)
Comment (stsepchanka) [reply src/middleware/authorize.ts]: Token parsing has been changed.

File: src/app.ts (hunk 3 above)
Comment (stsepchanka) [reply src/app.ts]: White list checking has been added.

**Pull request merged**